Todo.py

```diff
@@ -0,0 +1,203 @@
+TOKEN = "NTY4MDkyNTY5NTE5NzgzOTM2.XMiBDw.zCL0KhV_gOtxfyACctgYCLcLlEY"
+# PREFIX = "g:"
+
+# EMOJIS = {
+#     "wastebasket": "\U0001F5D1",
+#     "check": "\U00002705",
+#     "hourglass": "\U000023F3"
+# }
+
+#bot = commands.Bot(command_prefix=PREFIX)
+
+class Todo(commands.Cog):
+    """A suite of command to make a nice todo list."""
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        if message.author == bot.user:
+            return
+        await bot.process_commands(message)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, reaction):
+        if reaction.user_id != bot.user.id:
+            message = await bot.get_channel(reaction.channel_id).fetch_message(reaction.message_id)
+
+            with open(TODO_CHANNEL, "r") as file:
+                channelId = file.readlines()[0]
+
+            if reaction.channel_id == int(channelId): #Check if it's a todo-message (check if it's the good channel)
+                if len(message.embeds) > 0: # Check if it's an embed, I think this will avoid most problems
+                    if reaction.emoji.name == EMOJIS['wastebasket']:
+                        await bot.get_channel(reaction.channel_id).delete_messages([message])
+
+                        repostFieldValue = None
+                        for field in message.embeds[0].fields:
+                            if field.name == "Public repost":
+                                repostFieldValue = field.value
+                        
+                        if repostFieldValue != None:
+                            repostMessage = await bot.get_channel(int(repostFieldValue.split(':')[0][2:-2])).fetch_message(int(repostFieldValue.split(':')[1][1:]))
+                            await repostMessage.delete()
+                    elif reaction.emoji.name == EMOJIS['check']:
+                        await message.remove_reaction(EMOJIS['hourglass'], bot.user)
+                    elif reaction.emoji.name == EMOJIS['hourglass']:
+                        await message.remove_reaction(EMOJIS['check'], bot.user)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_remove(self, reaction):
+        message = await bot.get_channel(reaction.channel_id).fetch_message(reaction.message_id)
+
+        with open(TODO_CHANNEL, "r") as file:
+            channelId = file.readlines()[0]
+
+        if reaction.channel_id == int(channelId): # Check if it's a todo-message (check if it's the good channel)
+            if len(message.embeds) > 0: # Check if it's an embed, I think this will avoid most problems
+                if reaction.user_id != bot.user.id:
+                    if reaction.emoji.name == EMOJIS['check']:
+                        await message.add_reaction(EMOJIS['hourglass'])
+                    elif reaction.emoji.name == EMOJIS['hourglass']:
+                        await message.add_reaction(EMOJIS['check'])
+
+    @commands.group()
```

Comment:
shouldn't the use of this group of commands be restricted ?

Reply:
Yes ! It should :D

Reply:
Maybe you should add a new clearance level that could be configured ( see #4 ) ?